src/storage/disk_table.h

```diff
@@ -409,27 +409,33 @@ class DiskTable : public Table {
     bool IsExpire(const ::openmldb::api::LogEntry& entry) override;
 
     void CompactDB() {
-        for (rocksdb::ColumnFamilyHandle* cf : cf_hs_) {
+        for (rocksdb::ColumnFamilyHandle* cf : *cf_hs_) {
             db_->CompactRange(rocksdb::CompactRangeOptions(), cf, nullptr, nullptr);
         }
     }
 
     int CreateCheckPoint(const std::string& checkpoint_dir);
 
+    bool AddIndex(const ::openmldb::common::ColumnKey& column_key) override;
     bool DeleteIndex(const std::string& idx_name) override;
     uint64_t GetRecordIdxCnt() override;
     bool GetRecordIdxCnt(uint32_t idx, uint64_t** stat, uint32_t* size) override;
     uint64_t GetRecordPkCnt() override;
     uint64_t GetRecordByteSize() const override { return 0; }
     uint64_t GetRecordIdxByteSize() override;
 
+    std::vector<rocksdb::ColumnFamilyHandle*> GetCF_DS_() {
+        return *cf_hs_;
+    }
+
     int GetCount(uint32_t index, const std::string& pk, uint64_t& count) override; // NOLINT
 
  private:
     rocksdb::DB* db_;
     rocksdb::WriteOptions write_opts_;
     std::vector<rocksdb::ColumnFamilyDescriptor> cf_ds_;
-    std::vector<rocksdb::ColumnFamilyHandle*> cf_hs_;
+    std::shared_ptr<std::vector<rocksdb::ColumnFamilyHandle*>> cf_hs_;
+    std::mutex cf_hs_mutex;
```

Comment:
for `private` member, it is named with suffix `_`, e.g., `cf_hs_mutex_`

can use the same `mutex` for both `cf_hs_` and `cf_ds_`

Reply:
Get it.